### ISSUE TITLE
Metric showing only using a numpy function

### ISSUE THREAD

**Issue by unsignedrant** [keras-team/keras#16653]
[distill-bert-uncased.txt
](https://huggingface.co/distilbert-base-uncased/blob/main/vocab.txt)
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): custom
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS Monterey 12.4
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): v2.9.0-18-gd8ce9f9c301 2.9.1
- Python version: 3.10.4

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with:
python -c "import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"

**Describe the problem**.

`kendall_tau1` is a `tf` metric, and `kendall_tau2` is a numpy metric. For some reason the the first only shows `nan` and the second calculates the metric properly but is slow and inconvenient.

Describe the problem clearly here. Be sure to convey here why it's a bug in Keras or why the requested feature is needed.

**Describe the current behavior**.

Unless being calculated in a numpy function, the metric shows `nan`

**Describe the expected behavior**.

Should work?

**[Contributing](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md)**.

- Do you want to contribute a PR? (yes/no): no
- If yes, please read [this page](https://github.com/keras-team/keras/blob/master/CONTRIBUTING.md) for instructions
- Briefly describe your candidate solution(if contributing):

**Standalone code to reproduce the issue**.

    import random
    import string
    
    import pandas as pd
    import tensorflow as tf
    import tensorflow_addons as tfa
    import tensorflow_models as tfm
    from keras import Input, Model
    from keras.layers import Dense, Dropout
    from transformers import TFDistilBertModel
    
    
    def kendall_tau1(y_true, y_pred):
        metric = tfa.metrics.KendallsTau(0, 1)
        metric.update_state(y_true, y_pred)
        return metric.result()
    
    
    def kendall_tau_np(y_true, y_pred):
        return kendall_tau1(y_true, y_pred).numpy()
    
    
    def kendall_tau2(y_true, y_pred):
        return tf.numpy_function(kendall_tau_np, [y_true, y_pred], 'float32')
    
    
    def create_model(input_shape, bert_head):
        ids = Input(input_shape, dtype='int32', name='input_ids')
        masks = Input(input_shape, dtype='int32', name='attention_mask')
        x0 = dict(input_ids=ids, attention_mask=masks)
        x = bert_head(x0)[0]
        x = Dropout(0.2)(x)
        x = Dense(64)(x)
        x = Dropout(0.2)(x)
        output = Dense(1, 'sigmoid')(x)
        return Model(x0, output)
    
    
    if __name__ == '__main__':
        pretrained_model = 'distilbert-base-uncased'
        s = string.ascii_letters + ' '
        df = pd.DataFrame(
            {
                'text': [''.join(random.choice(s) for _ in range(10)) for _ in range(7)],
                'type': ['a', 'b', 'a', 'c', 'b', 'c', 'a'],
                'order': [3, 2, 1, 2, 1, 1, 2],
            }
        )
        df['order'] = df.groupby('type')['order'].rank(pct=True)
        groups = [*df.groupby('type')]
        max_length = len(max(groups, key=lambda x: x[1].shape[0])[1])
        tokenizer = tfm.nlp.layers.BertTokenizer(
            vocab_file='distilbert-base-uncased.txt',
            lower_case=True,
        )
        packer = tfm.nlp.layers.BertPackInputs(
            seq_length=max_length, special_tokens_dict=tokenizer.get_special_tokens_dict()
        )
        head = TFDistilBertModel.from_pretrained(
            pretrained_model,
            cache_dir='transformers-cache',
        )
        x = tokenizer(df['text'].values)
        x = packer(x)
        x['input_ids'] = x['input_word_ids']
        x['attention_mask'] = x['input_mask']
        del x['input_word_ids']
        del x['input_type_ids']
        del x['input_mask']
        model = create_model((None,), head)
        model.compile('adam', 'huber', metrics=[kendall_tau1, kendall_tau2])
        model.fit(x=x, y=df['order'].values)

Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to Colab/Jupyter/any notebook.

**Source code / logs**.

    1/1 [==============================] - 21s 21s/step - loss: 0.1512 - kendall_tau1: nan - kendall_tau2: 0.4763

Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by tilakrayal** [comment]
@unsignedrant,
Thank you for opening the issue. Looks like this issue is not related to Keras repo. Kindly post the issue in suitable repo for the resolution. Thank you!

**Comment by unsignedrant** [comment]
what is the suitable repo?

**Comment by tilakrayal** [comment]
@unsignedrant,
This issue is more related to `tensorflow-models` and `tensorflow-addons` repo. Could you please try posting the issue in those repo's. Thank you!

**Comment by unsignedrant** [comment]
Ok, thanks
